aws/data_source_aws_network_interface.go

```diff
@@ -132,20 +133,37 @@ func dataSourceAwsNetworkInterface() *schema.Resource {
 func dataSourceAwsNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
-	input := &ec2.DescribeNetworkInterfacesInput{
-		NetworkInterfaceIds: []*string{aws.String(d.Get("id").(string))},
+	filters, filtersOk := d.GetOk("filter")
+	id, idOk := d.GetOk("id")
+	if idOk && filtersOk {
+		return fmt.Errorf("Only one of filter or id can be assigned")
+	}
+
+	if !idOk && !filtersOk {
+		return fmt.Errorf("One of id or filter must be assigned")
+	}
+
+	input := &ec2.DescribeNetworkInterfacesInput{}
+	if idOk {
```

Comment:
We should probably check to ensure ID actually has a value `if idOk && id.(string) != "" {`